netket/nn/utils.py

```diff
@@ -66,12 +67,12 @@ def _to_array_rank(apply_fun, variables, σ_rank, n_states, normalize, stable, e
         norm2 = jnp.linalg.norm(psi_local) ** 2
         norm2, _ = mpi.mpi_sum_jax(norm2)
 
-        if exp:
-            psi_local /= jnp.sqrt(norm2)
-        else:
+        if log:
             log_psi_local -= 1 / 2 * jnp.log(norm2)
+        else:
+            psi_local /= jnp.sqrt(norm2)
 
-    if not exp:
+    if log:
         psi_local = log_psi_local
 
     psi, _ = mpi.mpi_allgather_jax(psi_local)
```

Comment:
@PhilipVinc is this MPI call going to work if ```log=True```?

Reply:
If I understand correctly, `allgather` only concatenates the array, and it has nothing to do with arithmetics like add or multiply.

Reply:
Exactly. There's no problem here